src/asynchronous/client.rs

```diff
@@ -100,40 +100,12 @@ impl Client {
                     }
                     res = receive(&mut reader) => {
                         match res {
-                            Ok((header, body)) => {
-                                tokio::spawn(async move {
-                                    let resp_tx2;
-                                    {
-                                        let mut map = req_map.lock().unwrap();
-                                        let resp_tx = match map.get(&header.stream_id) {
-                                            Some(tx) => tx,
-                                            None => {
-                                                debug!(
-                                                    "Receiver got unknown packet {:?} {:?}",
-                                                    header, body
-                                                );
-                                                return;
-                                            }
-                                        };
-
-                                        resp_tx2 = resp_tx.clone();
-                                        map.remove(&header.stream_id); // Forget the result, just remove.
-                                    }
-
-                                    if header.type_ != MESSAGE_TYPE_RESPONSE {
-                                        resp_tx2
-                                            .send(Err(Error::Others(format!(
-                                                "Recver got malformed packet {:?} {:?}",
-                                                header, body
-                                            ))))
-                                            .await
-                                            .unwrap_or_else(|_e| error!("The request has returned"));
-                                        return;
-                                    }
-
-                                    resp_tx2.send(Ok(body)).await.unwrap_or_else(|_e| error!("The request has returned"));
-                                });
+                            Ok((header, Ok(body))) => {
+                                spawn_trans_resp(req_map, header, Ok(body));
```

Comment:
why split this two cases: Ok((header, Ok(body)))  and Ok((header, Err(e))), since both of them were dealt with the same logic?

Reply:
fixed